Without this patch, the ganeti master will fail to connect to a node, with this in the node's /var/log/ganeti/node-daemon.log:
```
2020-04-21 11:21:34,253: ganeti-noded pid=10198 ERROR Error while handling request from 10.9.1.35:50222
Traceback (most recent call last):
  File "/usr/share/ganeti/2.15/ganeti/http/server.py", line 594, in _IncomingConnection
    self.request_executor(self, self.handler, connection, client_addr)
  File "/usr/share/ganeti/2.15/ganeti/server/noded.py", line 158, in __init__
    http.server.HttpServerRequestExecutor.__init__(self, *args, **kwargs)
  File "/usr/share/ganeti/2.15/ganeti/http/server.py", line 439, in __init__
    request_msg_reader, force_close)
  File "/usr/share/ganeti/2.15/ganeti/http/__init__.py", line 516, in ShutdownConnection
    write_timeout)
  File "/usr/share/ganeti/2.15/ganeti/http/__init__.py", line 417, in SocketOperation
    if isinstance(sock, OpenSSL.SSL.ConnectionType):
AttributeError: 'module' object has no attribute 'ConnectionType'
```

--- ./lib/http/__init__.py.orig
+++ ./lib/http/__init__.py
@@ -377,7 +377,7 @@
 
   # Handshake is only supported by SSL sockets
   if (op == SOCKOP_HANDSHAKE and
-      not isinstance(sock, OpenSSL.SSL.ConnectionType)):
+      not isinstance(sock, OpenSSL.SSL.Connection)):
     return
 
   # No override by default
@@ -414,7 +414,7 @@
           return sock.recv(arg1)
 
         elif op == SOCKOP_SHUTDOWN:
-          if isinstance(sock, OpenSSL.SSL.ConnectionType):
+          if isinstance(sock, OpenSSL.SSL.Connection):
             # PyOpenSSL's shutdown() doesn't take arguments
             return sock.shutdown()
           else:
